flict/flictlib/translator.py

```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
-        for item in translations:
-            #print(" let's check...." + str(item), end="")
-            if 'license_expression' in item and 'spdx_id' in item:
-                le = item['license_expression'].lower()
-                if "scancode" in le:
-                    logger.main_logger.debug(" IGNORING since scancode")
-                    pass
-                else:
-                    #print(" OK ")
-                    transl = item['spdx_id']
-                    key = item['license_expression']
-                    if transl not in symbols:
-                        symbols[transl] = []
-                    symbols[transl].append(key)
+    try:
+        with open(translations_file) as file_:
+            translations_object = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
+        raise RuntimeError("reading translations file failed")
+
+    translations = translations_object["translations"]
+    translations_map = {item['value']: item
+                        for item in translations
+                        if item.get('translation', False)}
+
+    for item in translations:
+        if 'license_expression' in item and 'spdx_id' in item:
+            le = item['license_expression'].lower()
+            if "scancode" in le:
+                logger.main_logger.debug(" IGNORING since scancode")
             else:
-                raise Exception("Failed parsing item:  " + str(item))
+                transl = item['spdx_id']
+                key = item['license_expression']
+                if transl not in symbols:
+                    symbols[transl] = []
+                symbols[transl].append(key)
+        else:
+            raise RuntimeError(f"Failed parsing item: {item}")
+    results = {'original': translations_object,
+               'translations_map': translations_map,
+               'symbols': symbols}
+    # enabling this crashes lots of tests :)
+    # return results
+    read_translations.results = results
     return symbols
 
 
 def read_packages_file(jsonfile, translations):
-    with open(jsonfile) as fp:
-        packages = json.load(fp)
-        # TODO: sync with flict (should be "package")
-        package = packages["component"]
-        deps = package["dependencies"]
-        license = package["license"].strip()
-        package["license"] = update_license(translations, license)
-        update_packages(translations, deps)
-        return packages
+    try:
+        with open(jsonfile) as file_:
+            packages = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
+        raise RuntimeError("reading packages file failed")
+    # TODO: sync with flict (should be "package")
+    package = packages["component"]
+    deps = package["dependencies"]
+    license = package["license"].strip()
+    package["license"] = update_license(translations, license)
+    update_packages(translations, deps)
+    return packages
+
+
+if __name__ == "__main__":
```

Comment:
I see that as a change of functionality - so it should come as a separate commit with some justification

Reply:
I'd say this is a bit of a gray area. As explained in the first comment, the file would would just add a different way to work with the file, without modifying how it was used from within flict. However, probably it might be more important to understand if this was ever the case, or could it be useful to work with it bypasing flict?

Reply:
@hesa could you provide some insights if this was ever meant to run standalone?